routing-daemon/lib/openshift/routing/models/f5-icontrol-rest.rb

```diff
@@ -235,11 +185,80 @@ def delete_pool_alias pool_name, alias_str
     end
 
     def add_ssl pool_name, alias_str, ssl_cert, private_key
-      # no-opp
-    end 
+      # The LTM must be configured with a default client SSL profile for fallback for custom alias SSL to work with SNI
+      # https://support.f5.com/kb/en-us/solutions/public/13000/400/sol13452.html
+
+      # write temp cert and key
+      certfname = "#{@tmpdir}/#{URI.escape(alias_str)}.crt"
+      keyfname = "#{@tmpdir}/#{URI.escape(alias_str)}.key"
```

Comment:
Not sure `URI.escape` is ideal.  I'm not sure there's really anything exploitable here either, but there's no need to be more trusting than necessary.  How about using `URI.escape "#{@tmpdir}/#{URI.escape(alias_str)}.crt", /[^a-zA-Z0-9._-]/'` for `certfname` and similar for `keyfname`? Alternatively, we could use something like `Shellwords.escape` and `gsub` to remove slashes, but it's easier just to whitelist alphanumeric and a few other characters.  For that matter, we could just use `Base64.encode64`.